src/rpc/handlers/LedgerEntry.cpp

```diff
@@ -97,11 +103,29 @@ LedgerEntryHandler::process(LedgerEntryHandler::Input input, Context const& ctx)
         auto const owner = util::parseBase58Wrapper<ripple::AccountID>(
             boost::json::value_to<std::string>(input.depositPreauth->at(JS(owner)))
         );
-        auto const authorized = util::parseBase58Wrapper<ripple::AccountID>(
-            boost::json::value_to<std::string>(input.depositPreauth->at(JS(authorized)))
-        );
+        // Only one of authorize or authorize_credentials MUST exist;
+        if (!input.depositPreauth->contains(JS(authorized)) &&
```

Comment:
Can we merge these two checks into one? Can below check work?
```
input.depositPreauth->contains(JS(authorized) == input.depositPreauth->contains(JS(authorize_credentials))
```

Reply:
I think it doesn't work if both are false.

Reply:
it works if both are false 😄